packages/react-instantsearch/examples/query-suggestions-rn/App.js

```diff
@@ -0,0 +1,251 @@
+          searchState={this.state.searchState}
+        >
+          <ConnectedSearchBox
+            displaySuggestions={this.displaySuggestions}
+            defaultRefinement={this.state.query}
+          />
+          <Index indexName="instantsearch_query_suggestions">
+            <Configure hitsPerPage={5} />
+            {suggestions}
+          </Index>
+          <Index indexName="instant_search">
+            <Configure hitsPerPage={15} />
+            <Text style={styles.bestResults}>Best results</Text>
+            <View>{results}</View>
+          </Index>
+        </InstantSearch>
+      </View>
+    );
+  }
+}
+
+class SearchBox extends Component {
+  render() {
+    return (
+      <View style={styles.searchBoxContainer}>
+        <Image
+          source={{
+            uri:
+              'https://d13yacurqjgara.cloudfront.net/users/1090953/avatars/small/3a0f064859092a0e82bedddcee24a4a8.png?148154278',
+          }}
+          style={styles.algoliaLogo}
+        />
+        <TextInput
+          style={styles.searchBox}
+          onChangeText={text => this.props.refine(text)}
+          value={this.props.currentRefinement}
+          placeholder={'Search a product...'}
+          placeholderTextColor={'black'}
+          clearButtonMode={'always'}
+          underlineColorAndroid={'white'}
+          spellCheck={false}
+          autoCorrect={false}
+          autoCapitalize={'none'}
+          onFocus={this.props.displaySuggestions}
+        />
+      </View>
+    );
+  }
+}
+
+const ConnectedSearchBox = connectSearchBox(SearchBox);
+
+SearchBox.propTypes = {
+  currentRefinement: PropTypes.string,
+  displaySuggestions: PropTypes.func,
+  refine: PropTypes.func,
+};
+
+const HitsList = ({ hits, removeSuggestions, onEndReached }) => (
+  <FlatList
+    renderItem={({ item }) => (
+      <View style={styles.hitsContainer}>
+        <Image
+          source={{
+            uri: `https://res.cloudinary.com/hilnmyskv/image/fetch/h_300,q_100,f_auto/${item.image}`,
+          }}
+          style={styles.hitsPicture}
+        />
+        <Text style={styles.hitsText}>
+          <Highlight
+            attributeName="name"
+            hit={item}
+            highlightProperty="_highlightResult"
+          />
+        </Text>
+      </View>
+    )}
+    data={hits}
+    keyExtractor={item => item.objectID}
+    onEndReached={onEndReached}
+    onScroll={removeSuggestions}
+    ItemSeparatorComponent={() => <View style={hits.hitsSeparator} />}
+  />
+);
+
+HitsList.propTypes = {
+  hits: PropTypes.array,
+  removeSuggestions: PropTypes.func,
+  onEndReached: PropTypes.func,
+};
+
+const ResultsInfiniteHits = connectInfiniteHits(
+  ({ hits, hasMore, refine, removeSuggestions }) => {
+    const onEndReached = function() {
+      if (hasMore) {
+        refine();
+      }
+    };
+    return (
+      <HitsList
+        removeSuggestions={removeSuggestions}
+        hits={hits}
+        onEndReached={onEndReached}
```

Comment:
we should probably revert the addition of `react/jsx-no-bind`, since this is clearly correct, even though eslint doesn't like it

cc @vvo @samouss

Reply:
If you convert the function to `() => {}` it woks I think (not sure). Or we can simply allow functions in the rule.

> https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md#allowfunctions